CMSgov/mint-app
# Easi 2265/confidentiality agreement

# EASI-2265

## Changes and Description

- NDA post login page for accepting agreement and displaying previous agreement date
- Collapsable banner for displaying NDA notice on routes / and /models
- Wrapper for all secure routes to redirect to NDA if not signed
- Cypress/Unit testing


![Screen Shot 2022-09-08 at 11 08 01 AM](https://user-images.githubusercontent.com/95709965/189158381-3ddfb5bb-03d0-46cd-b7e7-704944445942.png)

![Screen Shot 2022-09-08 at 11 08 29 AM](https://user-images.githubusercontent.com/95709965/189158498-359945b1-2027-44a8-92fb-3c01e1353e50.png)

## Notes

Banner initial state should be collapsed

## How to test this change

- Log into MINT with local auth (Okta test account EZOD does not yet have roles to accept agreement)
- Attempt to navigate to any route in MINT from url.  Should always direct you back to NDA until you have accepted
- Accept the agreement and continue to MINT
- Log out and re login with same user
- Verify that the checkbox is replaced with NDA accepted date

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
